src/pathfinding_service/service.py

```diff
@@ -285,56 +285,87 @@ def create_token_network_for_address(
-                chain_id=message.chain_id,
-            )
-            return
+        if message.canonical_identifier.chain_identifier != self.chain_id:
+            raise MessageException('Received Capacity Update with unknown chain identifier')
 
         # check if token network exists
-        token_network = self._get_token_network(message.token_network_address)
+        token_network = self._get_token_network(token_network_address)
         if token_network is None:
-            log.info(
-                'Received balance update with unknown token network',
-                token_network_address=message.token_network_address,
-            )
-            return
+            raise MessageException('Received Capacity Update with unknown token network')
 
         # check if channel exists
-        if message.channel_identifier not in token_network.channel_id_to_addresses:
-            log.info(
-                'Received balance update with unknown channel_id in token network',
-                channel_id=message.channel_identifier,
-                token_network_address=message.token_network_address,
+        channel_identifier = message.canonical_identifier.channel_identifier
+        if channel_identifier not in token_network.channel_id_to_addresses:
+            raise MessageException(
+                'Received Capacity Update with unknown channel identifier in token network',
             )
-            return
 
         # TODO: check signature of message
 
         # check values max int 256 TODO: get constants
         if not 0 < message.updating_capacity < 2 ^ 256:
-            return
+            raise MessageException('Received Capacity Update with impossible capacity')
         if not 0 < message.other_capacity < 2 ^ 256:
-            return
+            raise MessageException('Received Capacity Update with impossible capacity')
+
+        # check if participants fit to channel id
+        if message.updating_participant not in token_network.channel_id_to_addresses[
+            message.canonical_identifier.channel_identifier
+        ]:
+            raise MessageException(
+                'Sender of Capacity Update does not match the internal channel',
+            )
+        if message.other_participant not in token_network.channel_id_to_addresses[
+            message.canonical_identifier.channel_identifier
+        ]:
+            raise MessageException(
+                'Other Participant of Capacity Update does not match the internal channel',
+            )
+
+        # check if nonce is higher than current nonce
+
+        view_to_partner, view_from_partner = token_network.get_channel_view_perspective(
+            channel_identifier=message.canonical_identifier.channel_identifier,
+            updating_participant=message.updating_participant,
+            other_participant=message.other_participant,
+        )
+
+        if (
+            message.updating_nonce <= view_to_partner.update_nonce and
```

Comment:
Usually we put the conditions in an extra variable to make the condition easier readable.

Reply:
I do not quite understand what you mean. Maybe like this?
```
       a = view_to_partner.update_nonce
       b = view_from_partner.update_nonce
      if (
            message.updating_nonce <= a and
            message.other_nonce <= b
        ):
```

Reply:
More like:
```
      valid nonce = (
         message.updating_nonce <= view_to_partner.update_nonce and
         message.other_nonce <= view_from_partner.update_nonce
      if valid_nonce:

Reply:
understood, changed it